### PR TITLE
Azure Key Vault application variable provider support in Spin 2.5

### PR DESCRIPTION
**Pull request by ThorstenHans**
This PR adds documentation for using Azure Key Vault as application variable provider.

This should **not be merged before** the corresponding sample has been merged (https://github.com/fermyon/spin/pull/2483)

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title`, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result)
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors
- [x] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

**Comment by ThorstenHans** [comment]
@tpmccallum please note:  

I've moved both, the Vault Provider and the Azure Key Vault Provider sample apps to [this location in fermyon/enterprise-architectures-and-patterns](https://github.com/fermyon/enterprise-architectures-and-patterns/tree/main/application-variable-providers).

The most recent commit to this PR [0e5731e](https://github.com/fermyon/developer/pull/1268/commits/0e5731e5eed9f79d4ad99af637d1888635cf5fdd) updates corresponding links

File: content/spin/v2/dynamic-configuration.md [hunk 1]
```diff
@@ -113,6 +115,115 @@ $ curl localhost:3000 --data "wrong_password"
...
+vault_url = "https://spin.vault.azure.net/"
+client_id = "12345678-1234-1234-1234-123456789012"
+client_secret = "some.generated.password"
+tenant_id = "12345678-1234-1234-1234-123456789012"
+authority_host = "AzurePublicCloud"
+```
+
+#### Azure Key Vault Application Variable Provider Example
+
+1. Deploy Azure Key Vault
```
Comment (tpmccallum) [inline content/spin/v2/dynamic-configuration.md:143]:
```suggestion
1. Deploy Azure Key Vault:
```

File: content/spin/v2/dynamic-configuration.md [hunk 2]
```diff
@@ -113,6 +115,115 @@ $ curl localhost:3000 --data "wrong_password"
...
+$ az keyvault create -n $KV_NAME \
+  -g $RG_NAME \
+  -l $LOCATION \
+  --enable-rbac-authorization true
+
+# Grab the Azure Resource Identifier of the Azure Key Vault instance
+$ KV_SCOPE=$(az keyvault show -n $KV_NAME -g $RG_NAME -otsv --query "id")
+```
+
+2. Add a Secret to the Azure Key Vault instance
```
Comment (tpmccallum) [inline content/spin/v2/dynamic-configuration.md:164]:
```suggestion
2. Add a Secret to the Azure Key Vault instance:
```

File: content/spin/v2/dynamic-configuration.md [hunk 3]
```diff
@@ -113,6 +115,115 @@ $ curl localhost:3000 --data "wrong_password"
...
+
+# Assign the "Key Vault Secrets User" role to the SP
+# allowing it to read secrets from the Azure Key Vault instance
+$ az role assignment create --assignee $CLIENT_ID \
+  --role "Key Vault Secrets User" \
+  --scope $KV_SCOPE
+```
+
+4. Go to the [Azure Key Vault variable provider example](https://github.com/fermyon/enterprise-architectures-and-patterns/tree/main/application-variable-providers/azure-key-vault-provider) application.
+5. Replace Tokens in `runtime_config.toml`
```
Comment (tpmccallum) [inline content/spin/v2/dynamic-configuration.md:205]:
```suggestion
5. Replace Tokens in `runtime_config.toml`.
```

**Review by tpmccallum — APPROVED** [review]
This is super, @ThorstenHans!
Thank you.
Just a few punctuation nits and LGTM.

**Comment by tpmccallum** [comment]
Noticed the system never created a private preview, but it looks perfect on localhost preview:
Screenshot 2024-05-03 at 07 54 38

**Pull request merged**